src/Microsoft.NetCore.Analyzers/Core/Security/SourceTriggeredTaintedDataAnalyzerBase.cs

```diff
@@ -92,96 +93,45 @@ public override void Initialize(AnalysisContext context)
                                 operationAnalysisContext =>
                                 {
                                     IInvocationOperation invocationOperation = (IInvocationOperation)operationAnalysisContext.Operation;
-
-                                    if (!sourceInfoSymbolMap.IsSourceMethodFast(
-                                            invocationOperation.TargetMethod,
-                                            invocationOperation.Arguments,
-                                            out bool isSourceMethod,
-                                            out bool requiresPointsTo,
-                                            out bool requiresValueContent))
-                                    {
-                                        return;
-                                    }
-
```

Comment:
In the method IsSourceMethod(), the PointsToAnalysisResult and ValueContentAnalysisResult will be evaluated only when the target method is a source method that needs PointsToAnalysis or ValueContentAnalysis. And for method that is not even a source method, won't be added to do PropertySetAnalysis.

Reply:
The performance regression was from running PointsToAnalysis even for PointsToCheck callbacks that always returned true.

I'd prefer to fix that by having SourceInfo have a ```ImmutableHashSet<(MethodMatcher MethodMatcher, ImmutableHashSet<string> TaintedTargets)> TaintedMethods``` (instead of the current approach of putting always-returning-true PointsToChecks into TaintedMethodsNeedsPointsToAnalysis).

Reply:
1. I think the performance regression was mainly from running PointsToAnalysis for every single method.

2. And every time we run TaintedDataAnalysis, PointsToAnalysis will be run. Now that PointsToAnalysisResult will be cached, what's the matter we run PointsToAnalysis before TaintedDataAnalysis?

Reply:
You're right for 1, my mistake.

The DFA result caching is limited at the moment (see #2795), so I'd still want to avoid PointsToAnalysis when trying to evaluate whether to run TaintedDataAnalysis.

You could try measuring the build time of some large code bases before and after your change, and see if there's any noticeable difference.

Reply:
After I turned down the PointsToAnalysisResult cache, they do have noticeable difference. When enables PointsToAnalysisResult cache, there's no difference cause DisposeObjectsBeforeLosingScope basically caches the most part result of PointsToAnalysis.